ogr/factory.py

```diff
@@ -84,7 +84,7 @@ def get_project(
     mapping = service_mapping_update.copy() if service_mapping_update else {}
     custom_instances = custom_instances or []
     for instance in custom_instances:
-        mapping[instance.instance_url] = instance.__class__
+        mapping[instance.hostname] = instance.__class__
```

Comment:
Shouldn't the docstring be updated as well?

Reply:
Good point, but after checking the docstring, I don't think it's needed. This is more a bugfix than a change.

Reply:
I mean, the doc for `service_mapping_update` says `Custom mapping from service url (str) to service class.` and you are updating that with hostname to class mappings.

Reply:
Ah, I see... Thanks!

Reply:
OK, to preserve old docstring and functionality, I've made the functions to support both hostname and URL.

@nforro PTAL, thanks!